File: backend/src/models/Service.js
const mongoose = require('mongoose');

const serviceSchema = new mongoose.Schema(
  {
    salon_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Salon', required: true },
    name: { type: String, required: true, trim: true },
    description: { type: String, trim: true, default: '' },
    price: { type: Number, required: true, min: 0 },
    duration: { type: Number, required: true, min: 5, max: 480 },
    active: { type: Boolean, default: true },
  },
  { timestamps: true }
);

serviceSchema.index({ salon_id: 1, name: 1 }, { unique: true });

module.exports = mongoose.model('Service', serviceSchema);

File: mobile_app/lib/models/service.dart
class SalonService {
  SalonService({
    required this.id,
    required this.name,
    required this.price,
    required this.duration,
  });

  final String id;
  final String name;
  final num price;
  final int duration;

  factory SalonService.fromJson(Map<String, dynamic> json) {
    return SalonService(
      id: json['_id'] ?? json['id'],
      name: json['name'] ?? '',
      price: json['price'] ?? 0,
      duration: json['duration'] ?? 0,
    );
  }
}
